reascripts/common/libs/ExecProcess.lua

```diff
@@ -13,6 +13,39 @@ ExecProcess = {
     local o = { command = command }
     setmetatable(o, ExecProcess)
     return o
+  end,
+
+  via_tempfile = function(command)
+    local ep = ExecProcess.new(command)
+
+    function ep:run(timeout)
+      local tempfile = Tempfile:name()
+
+      local f = io.open(tempfile, "w")
+      if not f then
+        return nil, "Unable to open tempfile"
+      end
+
+      f:write(ep.command)
+      f:close()
+
+      if EnvUtil.is_windows() then
+        ep.command = 'cmd /c "' .. tempfile .. '"'
```

Comment:
do you think it's worth continuing down this path? i know we've talked about it a couple of times before in a couple of different contexts. guess i figured it was worth a spike - happy to punt it to a separate pr.

Reply:
We can move the prefix/suffix stuff to a separate PR, or address it later, but we definitely need the .cmd extension for this to work on Windows (even if we just concatenate it here).

Or did you mean running tempfile scripts in general? I think it's a good option to have, and it would make it possible to do redirection. The lack of redirection is a real pain. On the other hand, I think it's good to avoid using the shell unless it's really necessary.

Reply:
i guess i was maybe just looking for a scope check generally? i half-regretted even going this deep just based on wanting to provide a `reveal` command 🤷🏻‍♂️ lol

i'll make some changes to add `.cmd` to the generated filename and manually remove it in the windows case. i think that technically means a brief period of time until a separate prefix/suffix pr where in the meantime the internal state of `Tempfile` is:
- modified externally (if windows, append `.cmd` and update `Tempfile._names` accordingly)
- left inconsistent

(just thinking out loud)